---
title: "CDSeq: a complete deconvolution method using sequencing data"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{CDSeq: a complete deconvolution method using sequencing data}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.width = 10, 
  fig.height = 7,
  fig.align = 'center'
)
```




# Introduction
CDSeq is a complete deconvolution method using sequencing data. Simply put, CDSeq takes bulk RNA-Seq read counts data as input and estimates the cell-type-specific gene expression profiles (csGEPs) and sample-specific cell-type proportions (SSP). 



# Usage

## Input
   Input                            Description
---------------------------  ------------------------
bulk_data                    this is the bulk RNAseq read counts data. It is a G $\times$ M matrix or data frame, where G is the number of genes and M is the sample size.
beta                         hyperparameter for csGEP estimation. 
alpha                        hyperparameter for SSP estimation.
cell_type_number             number of cell types. It is an integer or a vector of varying integers.
mcmc_iterations              number of iterations for Gibbs sampler.
dilution_factor              dilution factor is a number used to dilute the input bulk data for faster computation. CDSeq will dilute the bulk data by *dilution_factor*, i.e. $\frac{\text{bulk_data}}{\text{dilution_factor}}$ will be analyzed. 
gene_subset_size             number of genes used for a block. 
block_number                 number of blocks.
cpu_number                   number of CPUs will be used. If null, CDSeq will detect the number of available cores for parallel computing.
gene_length                  effective length of the genes in bulk data. It is defined as *gene length - read length + 1*.
reference_gep                pure cell line gene expressions. 
print_progress_msg_to_file   indicator of printing the running process.

## Ouput {#output}
Output                Description
--------------------  ------------------------
estGEP                CDSeq-estimated cell-type-specific gene expression profiles. It is a G by T matrix where G denotes the number of genes and M is the sample size.
estProp               CDSeq-estimated sample-specific cell-type proportions. It is a M by T matrix where M is the sample size and T is the number of cell types.
cell_type_assignment  If refGEP is given, CDSeq will perform one-to-one cell types assignment for CDSeq-estimated cell types.
lgpst                 This is the log posterior values for each element in cell_type_number.
estT                  CDSeq-estimated number of cell types.
est_all               If cell_type_number is a vector, CDSeq will return all estimations for all elements in cell_type_vector.
parameters            The user-provided parameters for CDSeq. 
gibbsRunningTime      Time consumed for Gibbs sampler. 
processIDs            Process ID in the operating system for running CDSeq.

# Examples
## Installation
You could install CDSeq from Github by running the following
```{r installation, eval=FALSE}
install_github("kkang7/CDSeq_R_Package")
```

## Quick example
### When number of cell types is a scalar
```{r library}
library(CDSeq)
```
The followig code runs **CDSeq** using synthetic mixtures comes with the R package. The *mixtureGEP* consists of 40 samples with 500 genes. These mixtures are generated using six pure cell lines and random generated mixing proportions. **CDSeq** takes *mixtureGEP* as input and outputs the estimates of cell-type-specific gene expressions and sample-specific cell-type proportions. 
```{r example 1, eval=FALSE, cache=TRUE, results='hide'}
result1<-CDSeq(bulk_data =  mixtureGEP, 
              cell_type_number = 6, 
              mcmc_iterations = 1000, 
              dilution_factor = 1, 
              block_number = 1, 
              gene_length = as.vector(gene_length), 
              reference_gep = refGEP, 
              cpu_number = 1, 
              print_progress_msg_to_file = 0)
```

The output *result* is a list containing multiple entities as listed in [link1](#output). When *cell_type_number* is a scalar, **CDSeq** will only returnt the following values.
```{r output, eval=FALSE, echo=TRUE}
ls(result1)
```
We can compare the CDSeq-estimated cell-type-specific GEPs with the ground truth.
```{r fig1, echo=TRUE,cache=TRUE,eval=FALSE}
trueGEP <- true_GEP_rpkm[,result$cell_type_assignment]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result$estGEP[,i],trueGEP[,i],xlab = "CDSeq-estimated GEPs", ylab = "True GEPs")
    max_v <- max(c(result$estGEP[,i], trueGEP[,i]))
    lines(c(0,max_v),c(0,max_v), type = "l")
}

```

We then compare the CDSeq-estimated cell type proportions with group truth.
```{r fig2, echo=TRUE,cache=TRUE,eval=FALSE}
trueProp <- true_prop_cell[result$cell_type_assignment,]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result$estProp[i,],trueProp[i,],xlab = "CDSeq-estimated proportions", ylab = "True proportions")
    lines(c(0,1),c(0,1), type = "l")
}
```

For this dataset, CDSeq takes about 10 minutes.
```{r time, echo=TRUE,cache=FALSE,eval=FALSE}
result$gibbsRunningTime
```



### Estimate the number of cell types
```{r example 2, eval=FALSE, cache=TRUE, results='hide'}
result2<-CDSeq(bulk_data =  mixtureGEP, 
              cell_type_number = 2:10, 
              mcmc_iterations = 700, 
              dilution_factor = 1, 
              block_number = 1, 
              gene_length = as.vector(gene_length), 
              reference_gep = refGEP, 
              cpu_number = 8, # use multiple cores to save time 
              print_progress_msg_to_file = 0)

```
```{r, eval=FALSE}
cor(result2$estGEP,true_GEP_rpkm)
```

```{r fig3, echo=FALSE, eval=FALSE}
par(mar=c(1,1,1,1))
par(mfrow=c(6,6))
par(pty="s")
for (i in 1:6) {
  for (j in 1:6) {
        plot(result2$estGEP[,i],true_GEP_gene[,j],xlab = "CDSeq-estimated GEPs", ylab = "True GEPs")
    max_v <- max(c(result2$estGEP[,i], true_GEP_gene[,j]))
    lines(c(0,max_v),c(0,max_v), type = "l")
  }
}
```


```{r fig4, echo=FALSE, eval=FALSE}
trueProp <- true_prop_cell[result2$cell_type_assignment,]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result2$estProp[i,],trueProp[i,],xlab = "CDSeq-estimated proportions", ylab = "True proportions")
    lines(c(0,1),c(0,1), type = "l")
}
```

```{r fig5,echo=FALSE, eval=FALSE}
lgpst <- rep(0,9)
for (i in 1:9) {
  lgpst[i] <- result2$est_all[[i]]$lgpst
}
plot(2:10,lgpst,xlab = "number of cell typess", ylab = "log posterior")
points(6,lgpst[5],pch=16,col="red")
lines(2:10, lgpst)
lines(c(6,6),c(0,lgpst[5]),lty=2)
```


### Reduce-Recover strategy for speeding up
```{r example 3, eval=FALSE, cache=TRUE, results='hide'}
result3<-CDSeq(bulk_data =  mixtureGEP, 
              cell_type_number = 6, 
              mcmc_iterations = 700, 
              dilution_factor = 1, 
              block_number = 10,
              gene_subset_size = 100,
              gene_length = as.vector(gene_length), 
              reference_gep = refGEP, 
              cpu_number = 10, 
              print_progress_msg_to_file = 0)
```


```{r fig6, echo=FALSE, eval=FALSE}
trueGEP <- true_GEP_rpkm[,result3$cell_type_assignment]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result3$estGEP[,i],trueGEP[,i],xlab = "CDSeq-estimated GEPs", ylab = "True GEPs")
    max_v <- max(c(result3$estGEP[,i], trueGEP[,i]))
    lines(c(0,max_v),c(0,max_v), type = "l")
  }
```


```{r fig7, echo=FALSE, eval=FALSE}
trueProp <- true_prop_cell[result3$cell_type_assignment,]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result3$estProp[i,],trueProp[i,],xlab = "CDSeq-estimated proportions", ylab = "True proportions")
    lines(c(0,1),c(0,1), type = "l")
}
```


```{r example 4, eval=FALSE, cache=TRUE, results='hide'}
result4<-CDSeq(bulk_data =  mixtureGEP, 
              cell_type_number = 2:10, 
              mcmc_iterations = 1000, 
              dilution_factor = 1, 
              block_number = 5,
              gene_subset_size = 50,
              gene_length = as.vector(gene_length), 
              reference_gep = refGEP, 
              cpu_number = 10, 
              print_progress_msg_to_file = 0)
```

```{r fig8, echo=FALSE, eval=FALSE}
trueGEP <- true_GEP_rpkm[,result4$cell_type_assignment]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result4$estGEP[,i],trueGEP[,i],xlab = "CDSeq-estimated GEPs", ylab = "True GEPs")
    max_v <- max(c(result4$estGEP[,i], trueGEP[,i]))
    lines(c(0,max_v),c(0,max_v), type = "l")
  }
```


```{r fig9, echo=FALSE, eval=FALSE}
trueProp <- true_prop_cell[result4$cell_type_assignment,]
par(mar=c(4,4,1,1))
par(mfrow=c(2,3))
par(pty="s")
for (i in 1:6) {
    plot(result4$estProp[i,],trueProp[i,],xlab = "CDSeq-estimated proportions", ylab = "True proportions")
    lines(c(0,1),c(0,1), type = "l")
}
```


# Comparison with single cell RNA seq data
CDSeq can be viewed as an *in silico* single cell profiling of the bulk RNA-Seq data.

