<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>CDSeq: Complete deconvolution using sequencing data • CDSeq</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="CDSeq: Complete deconvolution using sequencing data">
<meta property="og:description" content="Estimate cell-type-specific GEPs and sample-specific cell-type proportions simultaneously.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CDSeq</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.8</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/CDSeq-vignette.html">CDSeq: a complete deconvolution method using sequencing data</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="cdseq-r-package" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#cdseq-r-package" class="anchor"></a>CDSeq R Package</h1></div>
<p>CDSeq is a complete deconvolution method for dissecting bulk RNA-Seq data. The input of CDSeq is, ideally, bulk RNA-Seq read counts (for example HTSeq read counts), and CDSeq will estimate, simultaneously, the cell-type-specific gene expression profiles and the sample-specific cell-type proportions, no reference of pure cell line GEPs or scRNAseq reference is needed for running CDSeq.</p>
<p>For example, if you have a bulk RNA-Seq data, <span class="math inline"><em>A</em><sub><em>G</em> × <em>M</em></sub></span>, which is a G by M matrix. G denotes the number of genes and M is the sample size, then CDSeq will output <span class="math inline"><em>B</em><sub><em>G</em> × <em>T</em></sub></span> and <span class="math inline"><em>C</em><sub><em>T</em> × <em>M</em></sub></span>, where T is the number of cell types, <span class="math inline"><em>B</em><sub><em>G</em> × <em>T</em></sub></span> is the estiamte of cell-type-specific GEPs and <span class="math inline"><em>C</em><sub><em>T</em> × <em>M</em></sub></span> is the estiamte of sample-specific cell-type proportions.</p>
<p>Importantly, you can ask CDSeq to estimate the number of cell types, i.e. T, by providing a vector of possible integer values for T. For example, if the user input for T is a vector, i.e. <span class="math inline"><em>T</em> = {2, 3, 4, 5, 6}</span>, then CDSeq will estimate the most likely number for T.</p>
<div id="versioning" class="section level2">
<h2 class="hasAnchor">
<a href="#versioning" class="anchor"></a>Versioning</h2>
<p>I follow the versioning scheme: <em>major</em>.<em>minor</em>.<em>patch</em>. In CDSeq package, <em>major</em>, <em>minor</em> and <em>patch</em> basically indicate the following types of changes:</p>
<ol>
<li>
<em>major</em>: changes on the underlying probabilistic modeling</li>
<li>
<em>minor</em>: changes on the numerical procedures or coding structure</li>
<li>
<em>patch</em>: others</li>
</ol>
</div>
<div id="version-108" class="section level2">
<h2 class="hasAnchor">
<a href="#version-108" class="anchor"></a>Version 1.0.8</h2>
<p>Bug fix. Add vignette.</p>
</div>
<div id="version-107" class="section level2">
<h2 class="hasAnchor">
<a href="#version-107" class="anchor"></a>Version 1.0.7</h2>
<p>Bug fix. Add more features to cellTypeAssignSCRNA function.</p>
</div>
<div id="version-106" class="section level2">
<h2 class="hasAnchor">
<a href="#version-106" class="anchor"></a>Version 1.0.6</h2>
<p>Add cellTypeAssignSCRNA function. This function can be used for cell type assignment using user provided single cell RNAseq data (with annotations).</p>
</div>
<div id="version-105" class="section level2">
<h2 class="hasAnchor">
<a href="#version-105" class="anchor"></a>Version 1.0.5</h2>
<p>Add cellTypeAssignMarkerGenes function. This function can be used for cell type assignment using user provided marker gene list.</p>
</div>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>install_github(“kkang7/CDSeq_R_Package”)</p>
</div>
<div id="scrnaseq-data-for-cdseq-estimated-cell-type-annotations" class="section level2">
<h2 class="hasAnchor">
<a href="#scrnaseq-data-for-cdseq-estimated-cell-type-annotations" class="anchor"></a>scRNAseq data for CDSeq-estimated cell type annotations</h2>
<p>We collected some public available scRNAseq data from <a href="https://www.nature.com/articles/s41586-020-2157-4" class="uri">https://www.nature.com/articles/s41586-020-2157-4</a> and put them in hdf5 format. The data are available upon request.</p>
</div>
<div id="known-issue-about-macos-installation" class="section level2">
<h2 class="hasAnchor">
<a href="#known-issue-about-macos-installation" class="anchor"></a>Known issue about MacOS installation</h2>
<p>It is possible for Mac users to run into some errors when install from source due to problems of Rcpp compiler tools. Follow the instruction here may help: <a href="https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/" class="uri">https://thecoatlessprofessor.com/programming/cpp/r-compiler-tools-for-rcpp-on-macos/</a></p>
</div>
<div id="contact" class="section level2">
<h2 class="hasAnchor">
<a href="#contact" class="anchor"></a>Contact</h2>
<p>email: <a href="mailto:kangkai0714@gmail.com" class="email">kangkai0714@gmail.com</a></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-3">GPL-3</a></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Kai Kang <br><small class="roles"> Maintainer </small>  </li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p>Developed by Kai Kang.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
